WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/PostingActivityUseCase.kt

```diff
@@ -55,11 +61,43 @@ class PostingActivityUseCase
     override fun buildUiModel(domainModel: PostingActivityModel): List<BlockListItem> {
         val items = mutableListOf<BlockListItem>()
         items.add(buildTitle())
-        val activityItem = postingActivityMapper.buildActivityItem(domainModel.months, domainModel.max)
-        items.add(activityItem)
+        val activityItem = postingActivityMapper.buildActivityItem(
+                domainModel.months,
+                domainModel.max
+        )
+        val activityItemWithContentDescription = addBlockContentDescriptions(activityItem)
+        items.add(activityItemWithContentDescription)
         return items
     }
 
+    private fun addBlockContentDescriptions(activityItem: ActivityItem): ActivityItem {
+        val blocks = mutableListOf<Block>()
+
+        activityItem.blocks.forEach { block ->
+            val descriptions = mutableListOf<String>()
+            block.boxes.filter { box -> box.boxType != BoxType.INVISIBLE && box.boxType != BoxType.VERY_LOW }
+                    .sortedByDescending { box -> box.boxType }
+                    .groupBy { box -> box.boxType }
+                    .forEach { entry ->
+                        val readableBoxType = entry.key.name.replace(
```

Comment:
I just realized this isn't localized so I will switch this out with an implementation that is.